dvc/utils/serialize/_py.py

```diff
@@ -0,0 +1,96 @@
+import ast
+
+from funcy import reraise
+
+from ._common import ParseError, _load_data
+
+
+class PYFileCorruptedError(ParseError):
+    def __init__(self, path, message="PY file structure is corrupted"):
+        super().__init__(path, message)
+
+
+def load_py(path, tree=None):
+    return _load_data(path, parser=parse_py, tree=tree)
+
+
+def parse_py(text, path):
+    """Parses text from .py file into Python structure."""
+    with reraise(SyntaxError, PYFileCorruptedError(path)):
+        tree = ast.parse(text, filename=path)
+
+    result = ast_tree_to_dict(tree)
+    return result
+
+
+def ast_tree_to_dict(tree):
+    """Parses ast trees to dict."""
+    result = {}
+    for _body in tree.body:
+        try:
+            if isinstance(_body, ast.Assign) or isinstance(
+                _body, ast.AnnAssign
+            ):
+                result.update(ast_assign_to_dict(_body))
+            elif isinstance(_body, ast.ClassDef):
+                result.update({_body.name: ast_tree_to_dict(_body)})
+            elif (
+                isinstance(_body, ast.FunctionDef) and _body.name == "__init__"
+            ):
+                result.update(ast_tree_to_dict(_body))
+        except ValueError:
+            continue
+        except AttributeError:
+            continue
+    return result
+
+
+def ast_assign_to_dict(assign):
+    result = {}
+
+    if isinstance(assign, ast.AnnAssign):
+        name = get_ast_name(assign.target)
+    elif len(assign.targets) == 1:
+        name = get_ast_name(assign.targets[0])
+    else:
+        raise AttributeError
+
+    if isinstance(assign.value, ast.Dict):
+        _dct = {}
+        for key, val in zip(assign.value.keys, assign.value.values):
+            _dct[get_ast_value(key)] = get_ast_value(val)
+        result[name] = _dct
+    elif isinstance(assign.value, ast.List):
+        result[name] = [get_ast_value(val) for val in assign.value.elts]
+    elif isinstance(assign.value, ast.Set):
+        values = [get_ast_value(val) for val in assign.value.elts]
+        result[name] = set(values)
+    elif isinstance(assign.value, ast.Tuple):
+        values = [get_ast_value(val) for val in assign.value.elts]
+        result[name] = tuple(values)
+    else:
+        result[name] = get_ast_value(assign.value)
+
+    return result
+
+
+def get_ast_name(target):
+    if hasattr(target, "id"):
+        result = target.id
+    elif hasattr(target, "attr"):
+        result = target.attr
+    else:
+        raise AttributeError
+    return result
+
+
+def get_ast_value(value):
+    if isinstance(value, ast.Num):
+        result = value.n
+    elif isinstance(value, ast.Str):
+        result = value.s
+    elif isinstance(value, ast.NameConstant):
+        result = value.value
+    else:
+        raise ValueError
```

Comment:
Note that `ast.{Num/Str/NameConstant}` has been deprecated in Python3.8, but still works on 3.8.

Reply:
Fine as-is for now though.